### PR TITLE
Enable MPI support

### PR DESCRIPTION
**Pull request by awvwgk**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes #21 

- creates low-priority variants for MPI enabled versions using MPICH and OpenMPI (non-MPI version is installed by default)
- does not add MPI support for OSX on ARM

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {% set name = "plumed" %}
 {% set version = "2.7.1" %}
+{% set build = 1 %}
+{% set mpi = mpi or "nompi" %}
```
Comment (GiovanniBussi) [inline recipe/meta.yaml:4]: @awvwgk Can you explain this line? Is mpi set to mpi or to nompi?

**Comment by GiovanniBussi** [comment]
@awvwgk thanks a lot!!! I had a question on one line (added in the review). Can you also explain how to choose the version that will be installed (nompi/openmpi/mpich) or point me to a documentation page where this is discussed? Thanks again

File: recipe/meta.yaml (hunk 1 above)
Comment (awvwgk) [reply recipe/meta.yaml:4]: This is only for the linter which usually does check the meta.yaml without first expanding the build matrix (here variants of mpi) defined in the conda_build_config.yaml

**Comment by awvwgk** [comment]
The conda-forge knowledge base is quite detailed on this topic: https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi

The mpi package on conda-forge is a virtual package which makes MPICH and OpenMPI mutually exclusive. With the build variants of conda-build defined in `conda_build_config.yaml` in the recipe directory we create three variants (`nompi`, `mpich`, `openmpi`), where only the MPI variants depend on MPI. To make the selection straight-forward for conda install, we increase the build number for the non-MPI build to be always preferred over the MPI variants.

The user has to explicitly install the MPI variant (see https://dftbplus-recipes.readthedocs.io/en/latest/introduction.html#installing-from-conda-forge)

```sh
conda install plumed  # always non-MPI version of plumed
conda install 'plumed=*=mpi_*'  # some MPI version of plumed
conda install mpich 'plumed=*=mpi_*'  # MPICH version of plumed
```

**Comment by GiovanniBussi** [comment]
> The conda-forge knowledge base is quite detailed on this topic: https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi
> 
> The mpi package on conda-forge is a virtual package which makes MPICH and OpenMPI mutually exclusive. With the build variants of conda-build defined in `conda_build_config.yaml` in the recipe directory we create three variants (`nompi`, `mpich`, `openmpi`), where only the MPI variants depend on the `mpi` metapackage. To make the selection straight-forward for conda install, we increase the build number for the non-MPI build to be always preferred over the MPI variants.
> 
> The user has to explicitly install the MPI variant:
> 
> ```shell
> conda install plumed  # always non-MPI version of plumed
> conda install 'plumed=*=mpi_*'  # some MPI version of plumed
> conda install mpich 'plumed=*=mpi_*'  # MPICH version of plumed
> ```

Amazing... thanks a lot!

**Comment by GiovanniBussi** [comment]
@awvwgk sorry for coming back to this, I have a little doubt. In the [documentation you pointed to](https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi) there are also these lines:
```
{% if mpi == 'nompi' %}
# prioritize nompi variant via build number
{% set build = build + 100 %}
{% endif %}
```
These are not present in the plumed recipe. Could you explain how the priority of nompi is controlled?

Thanks!

Giovanni

**Comment by awvwgk** [comment]
My bad, I forgot to add those lines to recipe, sorry about that. I just checked and an unspecified install will pick some plumed version and maybe even install some MPI in the same environment, unless there is a constraint present which prefers one of the three available versions.

Adding those three lines to the recipe would properly prioritize the non-MPI version. I can submit a patch to fix it.

**Pull request merged**